core/spot_termination.go

```diff
@@ -121,13 +122,13 @@ func (s *SpotTermination) getAsgName(instanceID *string) (string, error) {
 	}
 
 	result, err := s.asSvc.DescribeAutoScalingInstances(&asParams)
-
-	var asgName = ""
-	if err == nil {
-		asgName = *result.AutoScalingInstances[0].AutoScalingGroupName
+	if err != nil {
+		return "", err
+	} else if len(result.AutoScalingInstances) == 0 {
```

Comment:
This is from #388. @cristim if you merge that first I can rebase this change away.

Reply:
Rebased away.